packages/slate-commons/src/commands/isNodeEmpty.ts

```diff
@@ -12,5 +17,13 @@ export function isNodeEmpty(editor: Editor, node: Node): boolean {
         );
     }
 
+    if (options?.shouldTrim) {
+        const str = Node.string(node);
+
+        if (str.length && str.trim() === '') {
+            return true;
+        }
+    }
```

Comment:
Hm... I think there might be a problem here. `Editor.isEmpty` is smarter and does additional checks to cover void nodes (void nodes always generate an empty string for `Node.string()`, but cannot be considered empty).
 
The code here will return `true` for a void node, which I believe, can lead bugs. Right?

I suggest, at least, adding a check to return `false` for void elements:

```tsx
if (editor.isVoid(node)) {
    return false; 
}
```

Also `Node.string()` will also return an empty string for an image element with empty caption, which might lead to bugs too. So it's better to cover that.

Am I overthinking it?

Reply:
You are right, the logic of this function could be confusing for future usages. 
What about moving this logic into this file https://github.com/prezly/slate/blob/master/packages/slate-commons/src/commands/isEmptyParagraphElement.ts it works with paragraphs and asumes that the node is a text node?

Reply:
> What about moving this logic

Let's try.